libs/utils/src/exp_counter.rs

```diff
@@ -0,0 +1,151 @@
+/// An exhausting, injective counter with exponential search properties
+///
+/// The type implements an [`Iterator`] that yields numbers from the range
+/// from 0 to a pre-defined maximum.
+///
+/// * It is *exhausting* in that it visits all numbers between 0 and `max`.
```

Comment:
```suggestion
/// * It is *exhausting* in that it visits all numbers between `0..max`.
```

Perhaps use Range notation here, since I was wondering why doesn't the `ExpCounter::with_max(40)` in test find that `40` but all `0..40` instead.

Reply:
fixed.